var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -234,37 +356,61 @@ def cmake_args(self):
         # `~kokkos` option is there to prevent +kokkos +omp setting to use omp directly from here
         # Same applies for raja
         if ("+omp" in self.spec) and ("~kokkos" in self.spec) and ("~raja" in self.spec):
-            args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+            args.append("-DCMAKE_CXX_COMPILER=" + spack_cxx)
+            args.append("-DCMAKE_C_COMPILER=" + spack_cc)
             if "cuda_arch" in self.spec.variants:
                 cuda_arch_list = self.spec.variants["cuda_arch"].value
                 # the architecture value is only number so append sm_ to the name
-                cuda_arch = "sm_" + cuda_arch_list[0]
-                args.append("-DOFFLOAD= " + "NVIDIA:" + cuda_arch)
-            elif "amdgpu_target" in self.spec.variants:
+                cuda_arch = "cc" + cuda_arch_list[0]
```

Comment:
I have updated the comments, the logic is : 
```
any model with  : -DCUDA_ARCH = sm_XY
any model with  : -arch = cc_XY 
(special ) since thrust has CMAKE_CUDA_ARCHITECTURES, it requires just value
```